### PR TITLE
ML-446 Changes to restructure the metrics

### PR DESCRIPTION
**Pull request by akanksha1304**
Restructured the metrics so that there are separate metrics, statistical tests, and confidence interval calculation modules.

Metrics contains:
Kullback Leibler Divergence
Jensen Shannon Divergence
Hellinger Distance
Port in correlation metrics from PR
R2 Mcfadden
Distance NN Correlation
Distance CN Correlation

Statistical tests contain:
Binomial Distance test
KS test
permutation tests
Kruskal Wallis test
Bootstrap test

Confidence interval file contains:
Bootstrap Confidence Intervals
Binomial Proportion CI

### CODE REVIEW

File: src/synthesized_insight/metrics/statistical_tests.py [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+                                                  n_samples=1000)
+        else:
+            ts_distribution = bootstrap_binned_statistic((sr_a, sr_b),
+                                                         lambda x, y: self.metric_cls_obj(x, y),
+                                                         n_samples=1000)
+        p_value = self._bootstrap_pvalue(metric_value, ts_distribution)
+        return metric_value, p_value
+
+
+class PermutationTest(TwoColumnMetricTest):
```
Comment (jamied157) [inline src/synthesized_insight/metrics/statistical_tests.py:243]: shall we try and remove this?
```
class PermutationTest(_Metric):
```
and then write our own `__call__` and remove/replace `_check_column_types`

Comment (jamied157) [reply src/synthesized_insight/metrics/statistical_tests.py:243]: same with bootstrap

File: src/synthesized_insight/check.py [hunk 2]
```diff
@@ -141,6 +141,13 @@ def affine(self, sr: pd.Series) -> bool:
             return True
         return False
 
+    def binary(self, sr: pd.Series) -> bool:
```
Comment (jamied157) [inline src/synthesized_insight/check.py:144]: we need to decide whether to leave this here or move it as a custom check

**Review by jamied157 — CHANGES_REQUESTED** [review]

File: src/synthesized_insight/check.py (hunk 2 above)
Comment (akanksha1304) [reply src/synthesized_insight/check.py]: Doing a custom check for now

File: src/synthesized_insight/metrics/statistical_tests.py (hunk 1 above)
Comment (akanksha1304) [reply src/synthesized_insight/metrics/statistical_tests.py]: Modified as suggested

**Review by jamied157 — APPROVED** [review]
Thanks for making all the changes! LGTM!

**Pull request merged**